Sources/StreamChat/StateLayer/Chat.swift

```diff
@@ -1012,17 +1020,17 @@ extension Chat {
 
 // MARK: - Chat Client
 
-// TODO: Needs a better solution
 private extension ChatClient {
-    var attachmentQueueUploader: AttachmentQueueUploader {
-        backgroundWorkers.compactMap { $0 as? AttachmentQueueUploader }.first!
-    }
-    
-    var messageEditor: MessageEditor {
-        backgroundWorkers.compactMap { $0 as? MessageEditor }.first!
-    }
-
-    var messageSender: MessageSender {
-        backgroundWorkers.compactMap { $0 as? MessageSender }.first!
+    func backgroundWorker<T>(of type: T.Type) throws -> T {
+        if let worker = backgroundWorkers.compactMap({ $0 as? T }).first {
+            return worker
+        }
+        if currentUserId == nil {
+            throw ClientError.CurrentUserDoesNotExist()
+        }
+        if !config.isClientInActiveMode {
+            throw ClientError.ClientIsNotInActiveMode()
+        }
+        throw ClientError("Background worker of type \(T.self) is not set up")
```

Comment:
This is trying to follow the flow in `ChatClient` and giving the most accurate error message. Maybe, this should be moved to close to `createBackgroundWorkers`.